src/squidpy/datasets/_datasets.py

```diff
@@ -0,0 +1,278 @@
+]
+
+
+# =============================================================================
+# Hardcoded dataset name types
+# =============================================================================
+
+# 10x Genomics Visium datasets (adata_with_image type)
+VisiumDatasets = Literal[
+    # spaceranger version 1.1.0 datasets
+    "V1_Breast_Cancer_Block_A_Section_1",
+    "V1_Breast_Cancer_Block_A_Section_2",
+    "V1_Human_Heart",
+    "V1_Human_Lymph_Node",
+    "V1_Mouse_Kidney",
+    "V1_Adult_Mouse_Brain",
+    "V1_Mouse_Brain_Sagittal_Posterior",
+    "V1_Mouse_Brain_Sagittal_Posterior_Section_2",
+    "V1_Mouse_Brain_Sagittal_Anterior",
+    "V1_Mouse_Brain_Sagittal_Anterior_Section_2",
+    "V1_Human_Brain_Section_1",
+    "V1_Human_Brain_Section_2",
+    "V1_Adult_Mouse_Brain_Coronal_Section_1",
+    "V1_Adult_Mouse_Brain_Coronal_Section_2",
+    # spaceranger version 1.2.0 datasets
+    "Targeted_Visium_Human_Cerebellum_Neuroscience",
+    "Parent_Visium_Human_Cerebellum",
+    "Targeted_Visium_Human_SpinalCord_Neuroscience",
+    "Parent_Visium_Human_SpinalCord",
+    "Targeted_Visium_Human_Glioblastoma_Pan_Cancer",
+    "Parent_Visium_Human_Glioblastoma",
+    "Targeted_Visium_Human_BreastCancer_Immunology",
+    "Parent_Visium_Human_BreastCancer",
+    "Targeted_Visium_Human_OvarianCancer_Pan_Cancer",
+    "Targeted_Visium_Human_OvarianCancer_Immunology",
+    "Parent_Visium_Human_OvarianCancer",
+    "Targeted_Visium_Human_ColorectalCancer_GeneSignature",
+    "Parent_Visium_Human_ColorectalCancer",
+    # spaceranger version 1.3.0 datasets
+    "Visium_FFPE_Mouse_Brain",
+    "Visium_FFPE_Mouse_Brain_IF",
+    "Visium_FFPE_Mouse_Kidney",
+    "Visium_FFPE_Human_Breast_Cancer",
+    "Visium_FFPE_Human_Prostate_Acinar_Cell_Carcinoma",
+    "Visium_FFPE_Human_Prostate_Cancer",
+    "Visium_FFPE_Human_Prostate_IF",
+    "Visium_FFPE_Human_Normal_Prostate",
+]
+
+# AnnData datasets (.h5ad)
+AnnDataDatasets = Literal[
+    "four_i",
+    "imc",
+    "seqfish",
+    "visium_hne_adata",
+    "visium_hne_adata_crop",
+    "visium_fluo_adata",
+    "visium_fluo_adata_crop",
+    "sc_mouse_cortex",
+    "mibitof",
+    "merfish",
+    "slideseqv2",
+]
+
+# Image datasets (.tiff)
+ImageDatasets = Literal[
+    "visium_fluo_image_crop",
+    "visium_hne_image_crop",
+    "visium_hne_image",
+]
+
+# SpatialData datasets (.zarr)
+SpatialDataDatasets = Literal["visium_hne_sdata",]
+
+
+# =============================================================================
+# 10x Genomics Visium functions
+# =============================================================================
+
+
+def visium(
+    sample_id: VisiumDatasets,
+    *,
+    include_hires_tiff: bool = False,
+    base_dir: PathLike | None = None,
+) -> AnnData:
+    """
+    Download Visium `datasets <https://support.10xgenomics.com/spatial-gene-expression/datasets>`_ from *10x Genomics*.
+
+    Uses the unified downloader which supports S3 with fallback to 10x Genomics servers.
```

Comment:
I'm not sure if we should have that fallback - the reason for S3 was control over the data existence. Feels overengineered here? I don't expect AWS to fail much

Reply:
I agree, we could then get rid of that agent-spoofing